examples/gno.land/p/gnoland/commondao/commondao.gno

```diff
@@ -0,0 +1,235 @@
+}
+
+// New creates a new common DAO.
+func New(options ...Option) *CommonDAO {
+	dao := &CommonDAO{
+		members:  &addrset.Set{},
+		active:   avl.NewTree(),
+		finished: avl.NewTree(),
+	}
+	for _, apply := range options {
+		apply(dao)
+	}
+	return dao
+}
+
+// ID returns DAO's unique identifier.
+func (dao CommonDAO) ID() uint64 {
+	return dao.id
+}
+
+// Slug returns DAO's URL slug.
+func (dao CommonDAO) Slug() string {
+	return dao.slug
+}
+
+// Name returns DAO's name.
+func (dao CommonDAO) Name() string {
+	return dao.name
+}
+
+// Path returns the full path to the DAO.
+// Paths are normally used when working with hierarchical
+// DAOs and is created by concatenating DAO slugs.
+func (dao CommonDAO) Path() string {
+	parent := dao.Parent()
+	if parent != nil {
+		prefix := parent.Path()
+		if prefix != "" {
+			return prefix + PathSeparator + dao.slug
+		}
+	}
+	return dao.slug
+}
+
+// Parent returns the parent DAO.
+// Null can be returned when DAO has no parent assigned.
+func (dao CommonDAO) Parent() *CommonDAO {
+	return dao.parent
+}
+
+// TopParent returns the topmost parent DAO.
+// The top parent is the root of the DAO tree.
+func (dao *CommonDAO) TopParent() *CommonDAO {
+	parent := dao.Parent()
+	if parent != nil {
+		return parent.TopParent()
+	}
+	return dao
+}
+
+// Members returns the list of DAO members.
+func (dao CommonDAO) Members() *addrset.Set {
+	return dao.members
+}
+
+// ActiveProposals returns all active DAO proposals.
+func (dao CommonDAO) ActiveProposals() rotree.IReadOnlyTree {
+	return dao.active
+}
+
+// FinishedProposalsi returns all finished DAO proposals.
+func (dao CommonDAO) FinishedProposals() rotree.IReadOnlyTree {
+	return dao.finished
+}
+
+// Propose creates a new DAO proposal.
+func (dao *CommonDAO) Propose(creator std.Address, d ProposalDefinition) (*Proposal, error) {
+	id, ok := dao.genID.TryNext()
+	if !ok {
+		return nil, ErrOverflow
+	}
+
+	p, err := NewProposal(uint64(id), creator, d)
+	if err != nil {
+		return nil, err
+	}
+
+	key := makeProposalKey(p.ID())
+	dao.active.Set(key, p)
+	return p, nil
+}
+
+// GetActiveProposal returns an active proposal.
+func (dao CommonDAO) GetActiveProposal(proposalID uint64) (_ *Proposal, found bool) {
+	key := makeProposalKey(proposalID)
+	if v, ok := dao.active.Get(key); ok {
+		return v.(*Proposal), true
+	}
+	return nil, false
+}
+
+// GetFinishedProposal returns a finished proposal.
+func (dao CommonDAO) GetFinishedProposal(proposalID uint64) (_ *Proposal, found bool) {
+	key := makeProposalKey(proposalID)
+	if v, ok := dao.finished.Get(key); ok {
+		return v.(*Proposal), true
+	}
+	return nil, false
+}
+
+// Vote submits a new vote for a proposal.
+func (dao *CommonDAO) Vote(member std.Address, proposalID uint64, c VoteChoice) error {
+	// TODO: Support other voting options though the proposal definition
+	if c != ChoiceYes && c != ChoiceNo && c != ChoiceAbstain {
+		return ErrInvalidVoteChoice
+	}
```

Comment:
What if the user of this library wants to have more vote choices?

Reply:
Users will be able to define custom voting options 👍 

It has not been implemented yet, that's why is still mentioned in the TODO here and in the `Tally` method. It was not implemented yet because for the boards2 dApp is not a requirement.

Reply:
Implemented in d348db1